proof/refine/AARCH64/ArchStateRelationLemmas.thy

```diff
@@ -164,6 +164,14 @@ lemma ghost_relation_wrapper_machine_state_upd_id[StateRelation_R_assms,simp]:
    = ghost_relation_wrapper s s'"
   by simp
 
+(* interface lemma, can't be used in locale assumptions due to free type variable *)
+lemma valid_arch_obj'_valid[wp]:
+  assumes P: "\<And>P T p. \<lbrace>\<lambda>s. P (typ_at' T p s)\<rbrace> f \<lbrace>\<lambda>rv s. P (typ_at' T p s)\<rbrace>"
+  notes [wp] = hoare_vcg_all_lift hoare_vcg_imp_lift hoare_vcg_const_Ball_lift typ_at_lifts [OF P]
+  shows "\<lbrace>valid_arch_obj' ako\<rbrace> f \<lbrace>\<lambda>rv. valid_arch_obj' ako\<rbrace>"
```

Comment:
I'm not sure if this comment will still make sense when we've forgotten the context and it's not clear what it's an interface lemma for and what locale it isn't able to be used in.

Reply:
I'd appreciate your input here then. The comment is trying to anticipate the future where someone goes "oh this is requalified, why isn't it part of the Arch interface" and then rediscovering that it's not really possible (e.g. when we have to migrate to MCS).

The term "interface lemma" appears in quite a few places, including in AInvs as shorthand for "Arch interface lemma".

Reply:
Fair point with "interface lemma" being used consistently for this. I guess I'm ok with it as is, something about how it's written still feels a bit off to me but I can't think of anything better.

Reply:
ah, missed that one due to a race. will tweak after tests pass

Reply:
done now